### PR TITLE
feature/253 Router confirmation cancel triggering route event

### PR DESCRIPTION
**Pull request by masterdany88**

### CODE REVIEW

File: nalu/src/main/java/com/github/nalukit/nalu/client/internal/route/AbstractRouter.java [hunk 1]
```diff
@@ -550,6 +550,10 @@ public void onCancel() {
                                            false);
                               // clear loop detection list ...
                               loopDetectionList.clear();
+                              fireRouterStateEvent(RouterState.ROUTING_ABORTED,
```
Comment (philkohl) [inline nalu/src/main/java/com/github/nalukit/nalu/client/internal/route/AbstractRouter.java:553]: Please add a new RouteState called ROUTING_CANCELED_BY_USER

**Review by philkohl — CHANGES_REQUESTED** [review]
see comment

**Comment by masterdany88** [comment]
Ok Enum type added. Still We miss previous route path before You trigger routing.

**Review by FrankHossfeld — APPROVED** [review]

**Review by FrankHossfeld — APPROVED** [review]

**Comment by masterdany88** [comment]
Did You test it? In event There is wrong url.

**Comment by FrankHossfeld** [comment]
Yeap, tested it. It looks good. The cancel event get fired and the route inside the event is the route, that is canceled by the user. So, nothing wrong.

**Comment by FrankHossfeld** [comment]
see my comment inside the issue.

**Pull request merged**